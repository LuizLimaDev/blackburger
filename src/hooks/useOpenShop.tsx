export default function useOpenShop(date = new Date()) {
  const format = new Intl.DateTimeFormat("pt-BR", {
    dateStyle: "full",
    timeStyle: "long",
    timeZone: "America/Sao_Paulo",
  }).format(date);
  const weekDay = format.split(" ")[0];
  const hours = format.split(" ")[7];
  const currentHour = hours.split(":")[0];

  let isOpen: boolean = false;

  if (
    currentHour >= "18" &&
    currentHour <= "24" &&
    weekDay !== "segunda-feira"
  ) {
    isOpen = true;
  }

  return isOpen;
}
