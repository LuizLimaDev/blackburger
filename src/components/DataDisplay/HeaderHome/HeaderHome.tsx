import useOpenShop from "@/hooks/useOpenShop";
import Image from "next/image";

export default function HeaderHome() {
  const isOpen = useOpenShop();

  return (
    <div className="mx-4 mt-8 flex justify-between">
      <div className="flex items-center">
        <Image
          src="/ui/logo.png"
          alt="logo black burger"
          width={72}
          height={67}
          priority={true}
          className="h-auto w-auto"
          aria-label="logo black burger"
        />
        <h1 className="bb-title text-2xl capitalize">Black Burger</h1>
      </div>

      <div className="flex flex-col items-center">
        <Image
          src="/icons/notification.svg"
          alt="notificação"
          width={0}
          height={0}
          sizes="100vw"
          className="w-6 h-7 drop-shadow-bb-2"
          aria-label="imagem de notificação"
        />
        <p
          className={`
        ${isOpen ? "text-green-500" : "text-red-500"}
        font-lilita text-xs uppercase mt-1 drop-shadow-bb-2
        `}
          role="paragraph"
        >
          {isOpen ? "Aberto" : "Fechado"}
        </p>
      </div>
    </div>
  );
}
