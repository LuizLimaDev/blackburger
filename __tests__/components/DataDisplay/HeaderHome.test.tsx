import HeaderHome from "@/components/DataDisplay/HeaderHome/HeaderHome";
import useOpenShop from "@/hooks/useOpenShop";
import "@testing-library/jest-dom";
import { render, renderHook, screen } from "@testing-library/react";

jest.mock("../../../src/hooks/useOpenShop");
const MockUseOpenShop = useOpenShop as jest.MockedFunction<typeof useOpenShop>;

describe("expect the Header have your elements", () => {
  test("render the logo img of Black Burger", () => {
    render(<HeaderHome />);
    const img = screen.getByRole("img", { name: "logo black burger" });
    expect(img).toBeInTheDocument();
  });

  test("render the title Black Burger", () => {
    render(<HeaderHome />);
    const h1 = screen.getByRole("heading");
    expect(h1).toBeInTheDocument();
  });

  test("render the text Black Burger", () => {
    render(<HeaderHome />);
    const text = screen.getByText("Black Burger");
    expect(text).toBeInTheDocument();
  });

  test("render the Black Burger title style", () => {
    render(<HeaderHome />);
    const h1 = screen.getByRole("heading");
    expect(h1).toHaveClass("bb-title text-2xl capitalize");
  });

  test("render the notification img", () => {
    render(<HeaderHome />);
    const img = screen.getByRole("img", { name: "imagem de notificação" });
    expect(img).toBeInTheDocument();
  });

  test("render text Fechado if isOpen is false", () => {
    MockUseOpenShop.mockReturnValue(false);
    render(<HeaderHome />);

    const { result } = renderHook(() => useOpenShop());
    const paragraph = screen.getByRole("paragraph");
    const text = screen.getByText(/Fechado/);

    expect(result.current).toBeFalsy();
    expect(paragraph).toHaveClass("text-red-500");
    expect(text).toBeInTheDocument();
    expect(paragraph).toMatchSnapshot();
  });

  test("render text Abert if isOpen is true", () => {
    MockUseOpenShop.mockReturnValue(true);
    render(<HeaderHome />);

    const { result } = renderHook(() => useOpenShop());
    const paragraph = screen.getByRole("paragraph");
    const text = screen.getByText(/Aberto/);

    expect(result.current).not.toBeFalsy();
    expect(paragraph).toHaveClass("text-green-500");
    expect(text).toBeInTheDocument();
    expect(paragraph).toMatchSnapshot();
  });
});
