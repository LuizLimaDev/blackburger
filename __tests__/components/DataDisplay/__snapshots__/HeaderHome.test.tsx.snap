// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect the Header have your elements render text Abert if isOpen is true 1`] = `
<p
  class="
        text-green-500
        font-lilita text-xs uppercase mt-1 drop-shadow-bb-2
        "
  role="paragraph"
>
  Aberto
</p>
`;

exports[`expect the Header have your elements render text Fechado if isOpen is false 1`] = `
<p
  class="
        text-red-500
        font-lilita text-xs uppercase mt-1 drop-shadow-bb-2
        "
  role="paragraph"
>
  Fechado
</p>
`;
